react/src/contexts/fetchCache/FetchCacheProvider.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { useEffect, useState } from "react";
+import { ClientContext, ResultContext } from "./FetchCacheContext";
+
+/**
+ * Provider for useFetchCache hook.
+ */
+export function FetchCacheProvider(props: { children: React.ReactNode }) {
+    const [url, setURL] = useState("");
```

Comment:
Naively thinking, shouldn't url not be a state and just be a prop? Isn't this the cause of the problem you mentioned or is that not how contexts work?

Reply:
I made the url a state variable because I knew that calling `setURL` would trigger a re-render and cause the `useEffect` code to run. 

The issue I had initially was that I stored only one cached result in a state variable as opposed to storing the entire cache. Storing one cached result caused an issue where if you ask for resource A, and then ask for resource B, the provider would give you resource A until it finished fetching B, causing undesirable behaviour. Now when you ask for A, it will return cache[A], which may be undefined indicating that it is not fetched yet.

Reply:
Ah, so the comment below is justifying the `useEffect`, not saying that the problem exists now?

Reply:
Yep. The one hook can handle multiple urls now.